Implement --print-cache-file option to print the path to the
cache file used by gdk-pixbuf.

--- docs/reference/gdk-pixbuf/gdk-pixbuf-query-loaders.1.orig	2010-11-18 12:33:06.511901670 +0100
+++ docs/reference/gdk-pixbuf/gdk-pixbuf-query-loaders.1	2010-11-18 12:34:15.216524449 +0100
@@ -31,7 +31,7 @@
 gdk-pixbuf-query-loaders \- GdkPixbuf loader registration utility
 .SH "SYNOPSIS"
 .HP \w'\fBgdk\-pixbuf\-query\-loaders\fR\ 'u
-\fBgdk\-pixbuf\-query\-loaders\fR [\-\-update\-cache] [module...]
+\fBgdk\-pixbuf\-query\-loaders\fR [\-\-update\-cache | \-\-print\-cache\-file ] [module...]
 .SH "DESCRIPTION"
 .PP
 
@@ -59,6 +59,11 @@ gdk\-pixbuf
 at the file\&.
 .SH "OPTIONS"
 .PP
+\-\-print\-cache\-file
+Prints the full path to the default cache file to stdout.
+.RS 4
+.RE
+.PP
 \-\-update\-cache
 .RS 4
 Write the output to the default cache location instead of
--- gdk-pixbuf/queryloaders.c.orig	2010-11-18 12:34:59.871657960 +0100
+++ gdk-pixbuf/queryloaders.c	2010-11-18 12:35:40.334628205 +0100
@@ -283,7 +283,11 @@ int main (int argc, char **argv)
 
 #endif
 
-        if (argc > 1 && strcmp (argv[1], "--update-cache") == 0) {
+	if (argc > 1 && strcmp (argv[1], "--print-cache-file") == 0) {
+		cache_file = gdk_pixbuf_get_module_file ();
+		printf ("%s\n", cache_file);
+		return 0;
+	} else if (argc > 1 && strcmp (argv[1], "--update-cache") == 0) {
                 cache_file = gdk_pixbuf_get_module_file ();
                 first_file = 2;
         }
